src/lib.rs

```diff
@@ -1054,6 +1069,76 @@ where
     }
 }
 
+// Borsh is byte-size little-endian de-needs-external-schema no-bit-compression serde.
+// Current ser/de for it is not optimal impl because const math is not stable nor primitives has bits traits.
+// Uses minimal amount of bytes to fit needed amount of bits without compression (borsh does not have it anyway).
+#[cfg(feature = "borsh")]
+impl<T, const BITS: usize> BorshSerialize for UInt<T, BITS>
+where
+    Self: Number,
+    T: BorshSerialize
+        + From<u8>
+        + BitAnd<T, Output = T>
+        + TryInto<u8>
+        + Copy
+        + Shr<usize, Output = T>,
+    <UInt<T, BITS> as Number>::UnderlyingType:
+        Shr<usize, Output = T> + TryInto<u8> + From<u8> + BitAnd<T>,
+{
+    fn serialize<W: borsh::io::Write>(&self, writer: &mut W) -> borsh::io::Result<()> {
+        let value = self.value();
+        let length = (BITS + 7) / 8;
+        let mut bytes = 0;
+        let mask: T = u8::MAX.into();
+        while bytes < length {
+            let le_byte: u8 = ((value >> (bytes << 3)) & mask)
```

Comment:
This can be simplified:

```
let le_bytes = (value >> (bytes << 3)) as u8;
```

No need for the expect, try_into() etc

Reply:
value is not primitive, so 'as' does not work as i tried

Reply:
could use https://docs.rs/num/latest/num/traits/trait.AsPrimitive.html , but do not want to force num traits usage